### PR TITLE
React implementation for map layer admin

### PR DESCRIPTION
**Pull request by atlain**
TODO
- Styling. Some components have problems when used inside a flyway. Probably because of the z-index used. These include at least Select, Slider (labels) and Confirm popup
- Permissions tab

### CODE REVIEW

File: bundles/admin/admin-layereditor/components/Opacity.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        this.props.onChange(value);
+    }
+
+    // Inline styles could be handled in some other way
+    // Slider component had some problems if styled-components was used
+    render () {
+        const { opacity } = this.state;
+        return (
+            <div style={{width: 500}}>
+                <div style={{float: 'left', width: 350, marginLeft: 10, marginRight: 10}}>
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/components/Opacity.jsx]: Use styled-components?

File: bundles/admin/admin-layereditor/components/Opacity.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    <Slider
+                        min={0}
+                        max={100}
+                        onChange={this.onChange}
+                        value={typeof opacity === 'number' ? opacity : 0}
+                    />
+                </div>
+                <div style={{float: 'left', width: 120}}>
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/components/Opacity.jsx]: Use styled-components?

File: bundles/admin/admin-layereditor/components/UrlInput.jsx [hunk 3]
```diff
@@ -53,6 +59,7 @@ export class UrlInput extends React.Component {
         });
     }
     render () {
+        const {credentials, ...other} = {...this.props};
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/components/UrlInput.jsx]:
```suggestion
        const {credentials, ...other} = this.props;
```

File: bundles/admin/admin-layereditor/components/UrlInput.jsx [hunk 4]
```diff
@@ -63,19 +70,36 @@ export class UrlInput extends React.Component {
                 ))}
             </Select>
         );
-
+        const credentialProps = {...credentials};
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/components/UrlInput.jsx]: Why are we shallow copying?

File: bundles/admin/admin-layereditor/view/AdminLayerForm.jsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import {Alert} from '../components/Alert';
+import {GenericContext} from '../../../../src/react/util.jsx';
+export class AdminLayerForm extends React.Component {
+    constructor ({layer, dataProviders, mapLayerGroups, flyout}) {
+        super();
+        this.flyout = flyout;
+        this.service = new AdminLayerFormService(() => this.setState({ layer: this.service.getLayer() }));
+        this.service.initLayerState(layer, mapLayerGroups, dataProviders);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleCancel = this.handleCancel.bind(this);
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm.jsx]: Use handleBinder?

File: bundles/admin/admin-layereditor/view/AdminLayerForm.jsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import {AdditionalTabPane} from './AdminLayerForm/AdditionalTabPane';
+import {PermissionsTabPane} from './AdminLayerForm/PermissionsTabPane';
+import {AdminLayerFormService} from './AdminLayerForm/AdminLayerFormService';
+import {StyledRoot} from './AdminLayerForm/AdminLayerFormStyledComponents';
+import {Alert} from '../components/Alert';
+import {GenericContext} from '../../../../src/react/util.jsx';
+export class AdminLayerForm extends React.Component {
+    constructor ({layer, dataProviders, mapLayerGroups, flyout}) {
+        super();
+        this.flyout = flyout;
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm.jsx]: We should keep the flyout instance out of the React component. Pass props `flyoutExists` and ´onCancel´ instead.

File: bundles/admin/admin-layereditor/view/AdminLayerForm/GeneralTabPane.jsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+import {UrlInput} from '../../components/UrlInput';
+import {Collapse, Panel} from '../../components/Collapse';
+import {MapLayerGroups} from './MapLayerGroups';
+import {StyledTab, StyledComponentGroup, StyledComponent} from './AdminLayerFormStyledComponents';
+import {GenericContext} from '../../../../../src/react/util.jsx';
+
+export const GeneralTabPane = ({layer, service}) => {
+    return (
+        <GenericContext.Consumer>
+            {value => {
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm/GeneralTabPane.jsx]:
```suggestion
            {({loc}) => {
```

File: bundles/admin/admin-layereditor/view/AdminLayerForm/GeneralTabPane.jsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                const loc = value.loc;
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm/GeneralTabPane.jsx]:
```suggestion
```

File: bundles/admin/admin-layereditor/view/AdminLayerForm/MapLayerGroups.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    const dataSource = allGroups.map((group) =>
+        <Checkbox key={group.id} onChange={(evt) => service.setMapLayerGroup(evt.target.checked, group.id)} checked={group.checked}>{group.name[lang]}</Checkbox>
+    );
+    const renderItem = (item) => {
+        return (
+            <ListItem>{item}</ListItem>
+        );
+    };
+    return (
+        <GenericContext.Consumer>
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm/MapLayerGroups.jsx]: use withContext()?

File: bundles/admin/admin-layereditor/view/AdminLayerForm/AdminLayerFormService.js [hunk 10]
```diff
@@ -0,0 +1,227 @@
+import queryString from 'query-string';
```
Comment (data-ux) [inline bundles/admin/admin-layereditor/view/AdminLayerForm/AdminLayerFormService.js]: Import only `stringify` function if nothing else is needed.

**Comment by jheijari** [comment]
Closing this PR. Development continued in https://github.com/oskariorg/oskari-frontend/pull/942.

**Pull request closed without merging**